api/observability/v1/clusterlogforwarder_types.go

```diff
@@ -171,16 +171,8 @@ type SecretKey struct {
 // BearerToken allows configuring the source of a bearer token used for authentication.
 // The token can either be read from a secret or from a Kubernetes ServiceAccount.
 type BearerToken struct {
-	// Name of the key used to get the value from the referenced Secret.
-	//
-	// +kubebuilder:validation:Optional
-	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Key Name",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
-	Key string `json:"key,omitempty"`
-
-	// Use Secret if the value should be sourced from a Secret in the same namespace.
-	//
 	// +kubebuilder:validation:Optional
-	Secret *corev1.LocalObjectReference `json:"secret,omitempty"`
+	*SecretKey `json:",inline"`
```

Comment:
As per slack comments I think this should be
```
Secret *SecretKey `json:"secret, omitempty`
```
Slack comment https://redhat-internal.slack.com/archives/C03744G7KE1/p1715294246066419?thread_ts=1715267662.470479&cid=C03744G7KE1

Reply:
I think we can leave it as is for now with @jcantrill 's auth changes.